### PR TITLE
Added Dark Mode support to web app

### LINKED ISSUE #5: Add support for dark mode
## User story

As a transit rider waiting for a bus at night,
I want the web app to respect my device's light/dark mode preferences,
so that my user experience is more pleasant.

## Requirements

- [x] Implement proper support for Dark Mode using Tailwind's `dark` variant: https://tailwindcss.com/docs/dark-mode
- [x] Implement proper support for Google Maps's Night Mode: https://developers.google.com/maps/documentation/javascript/examples/style-array
- [x] When my device is in light mode, I should see a light-styled UI
- [x] When my device is in dark mode, I should see a dark-styled UI

### PR DESCRIPTION
**Pull request by tarunsinghofficial**
Fixes #5

**Tasks done:**
- Implemented proper support for Dark Mode using Tailwind's dark variant.
- Ensured the map theme persisted based on user preference (light/dark) upon page reload.
- The spinner displays while bus stop data is being fetched, improving the user experience by indicating loading status.
- Ensured consistent theme handling across different components.

**Screenshot:**

bandicam2024-07-0120-08-45-733-ezgif com-crop

### CODE REVIEW

File: src/components/navigation/Header.svelte [hunk 1]
```diff
@@ -1,9 +1,15 @@
 <script>
     import { PUBLIC_OBA_REGION_NAME } from '$env/static/public';
+    import ThemeSwitcher from '$lib/ThemeSwitch/ThemeSwitcher.svelte';
+
+    import '../../app.css';
```
Comment (aaronbrethorst) [inline src/components/navigation/Header.svelte:5]: isn't this already included in the application layout?

**Review by aaronbrethorst — CHANGES_REQUESTED** [review]
Works great! However, can you undo the indentation changes? it makes it hard to figure out what has and hasn't changed in files like GoogleMap.svelte. Also, did you run a formatter over these files or something? I'd love to make sure that we're using consistent formatting across the project.

**Comment by tarunsinghofficial** [comment]
> Works great! However, can you undo the indentation changes? it makes it hard to figure out what has and hasn't changed in files like GoogleMap.svelte. Also, did you run a formatter over these files or something? I'd love to make sure that we're using consistent formatting across the project.

I think my prettier vs code extension formatted these files and it gets auto saved. I'll try reformatting it back to previous version.

**Comment by aaronbrethorst** [comment]
Ahh, I see. Thank you! I’ll spend some time getting linting and formatting set up as a precondition for merging so that we can better guarantee that our formatting choices align :)

**Comment by tarunsinghofficial** [comment]
> Ahh, I see. Thank you! I’ll spend some time getting linting and formatting set up as a precondition for merging so that we can better guarantee that our formatting choices align :)

Oh great! Thanks for it.

**Comment by tarunsinghofficial** [comment]
> Ahh, I see. Thank you! I’ll spend some time getting linting and formatting set up as a precondition for merging so that we can better guarantee that our formatting choices align :)

For now, I have manually pushed the changes after undoing all the indentation changes. Let me know if it's correct! Thanks.

File: src/components/navigation/Header.svelte (hunk 1 above)
Comment (tarunsinghofficial) [reply src/components/navigation/Header.svelte]: Yeah, removed from the Header.svelte

**Review by aaronbrethorst — APPROVED** [review]
nice work Tarun 👍

**Pull request merged**